### PR TITLE
HPCC-16089 Fix trigraph warning in esp logging code

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@rpastrana please review.

File: esp/logging/loggingagent/espserverloggingagent/loggingagent.hpp [hunk 1]
```diff
@@ -54,7 +54,12 @@ class CTransIDBuilder : public CInterface, implements IInterface
         if (value)
             id.append(value->get()).append('-');
         else
-            id.append('??-');
+        {
+            const char* ptr = key;
+            if (strlen(key) > 11) //skip 'Transaction'
```
Comment (rpastrana) [inline esp/logging/loggingagent/espserverloggingagent/loggingagent.hpp]: we might want to expand upon the comment... skiping the "transaction" prefix of the key

Comment (rpastrana) [inline esp/logging/loggingagent/espserverloggingagent/loggingagent.hpp]: Also how can we be sure the key will always contain the prefix 'Transaction' ? at the the very least we might want to make a comment where ever the keys are defined alerting the fact that 'TransactionXYZ' is expected?

**Comment by rpastrana** [comment]
@wangkx couple of comments

**Comment by wangkx** [comment]
@rpastrana please review my changes.

**Comment by rpastrana** [comment]
@wangkx looks reasonable

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**